### PR TITLE
Update jupyter and enable a few tests

### PR DESCRIPTION
**Pull request by globin**
cc @FRidh 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/notebook/default.nix [hunk 1]
```diff
@@ -21,32 +22,39 @@
...
   name = "${pname}-${version}";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "1cea3bbbd03c8e5842a1403347a8cc8134486b3ce081a2e5b1952a00ea66ed54";
+    sha256 = "1sh3jkkmjzv17c3r8ii3kfhpxi6dkjk846b2lfy71g9qwqdcvbvz";
   };
 
-  LC_ALL = "en_US.UTF-8";
+  LC_ALL = "en_US.utf8";
```
Comment (Mic92) [inline pkgs/development/python-modules/notebook/default.nix]: Is `en_US.utf8` a valid identifier?

Comment (globin) [reply pkgs/development/python-modules/notebook/default.nix]: $ locale -a    
C
en_GB.utf8
en_US.utf8
POSIX

Comment (Mic92) [reply pkgs/development/python-modules/notebook/default.nix]: Interesting on ubuntu/archlinux always the first notation is used. Python seems to accept both variants:

```console
$ LC_ALL=en_US.utf8 python -c 'import locale; print(locale.getdefaultlocale())'
('en_US', 'UTF-8')
$ LC_ALL=en_US.UTF-8 python -c 'import locale; print(locale.getdefaultlocale())'
('en_US', 'UTF-8')
```

**Comment by FRidh** [comment]
Thanks!

**Comment by TikhonJelvis** [comment]
@Fuuzetsu I also just ran into build failures for Jupyter on OS X. I suspect it's caused by this PR.

Is there a dedicated issue for that? Is it worth opening one?

**Comment by Fuuzetsu** [comment]
I haven't opened one. I have set doCheck to false in own checkout at the time to get around that. We should create an issue and/or just disable to test on OSX in the expression.

**Pull request merged**